clang/test/AST/ast-dump-template-decls-json.cpp

```diff
@@ -217,21 +221,21 @@ void i();
 // CHECK-NEXT:    "id": "0x{{.*}}",
 // CHECK-NEXT:    "kind": "FunctionTemplateDecl",
 // CHECK-NEXT:    "loc": {
-// CHECK-NEXT:     "offset": {{[0-9]+}},
+// CHECK-NEXT:     "offset": 142,
```

Comment:
It looks like a regular expression was previously used to avoid sensitivity to the `offset` values. I'm not familiar with this test, but I see the comment above regarding using `gen_ast_dump_json_test.py` to generate the `CHECK` lines. Did you use that script?

Reply:
Yes I did. All changes in this file where made by the script.

Reply:
Hmm, the regex changes were made in https://github.com/llvm/llvm-project/commit/6d8e5c968538eff2c024d46d5715de0828a69407 by Chris Bieneman and reviewed by Aaron in https://reviews.llvm.org/D119362. It looks like those regex changes need to be preserved. It seems that script could use an update. Perhaps follow up with Chris?

Reply:
It looks like all the script does is dump the output of `-ast-dump=json` without really checking the string. Theoretically I guess it would be possible to check for "offset" in this string and modifying what is generated, but I haven't worked with Python in years and so might not be the best person to do this update. I think this should also be done in a separate patch since it falls outside the scope of this patch. 

For now I can revert the offset changes the script made and just keep the new CHECK lines with offsets modified? Would you be ok with that? Tagging @llvm-beanz for any input as well.

Reply:
I think we just forgot to update that script to produce regexes here; I think it's fine to revert back to the regex and fix the script in a follow-up (doesn't have to be @elizabethandrews doing the follow-up either).

Reply:
To add a bit of context on the change to using regex matches. These tests are fragile to your system’s native line-endings. I changed these to regex matches in an effort to get clang’s tests to pass under a git checkout with autocrlf enabled.

Reply:
Reverting the undesired changes suffices for me. My Python experience is years out of date too :)